test_conformance/basic/test_loop.cpp

```diff
@@ -21,165 +21,117 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 
+#include <vector>
 
 #include "procs.h"
 
-const char *loop_kernel_code =
-"__kernel void test_loop(__global int *src, __global int *loopindx, __global int *loopcnt, __global int *dst)\n"
-"{\n"
-"    int  tid = get_global_id(0);\n"
-"    int  n = get_global_size(0);\n"
-"    int  i, j;\n"
-"\n"
-"    dst[tid] = 0;\n"
-"    for (i=0,j=loopindx[tid]; i<loopcnt[tid]; i++,j++)\n"
-"    {\n"
-"        if (j >= n)\n"
-"            j = 0;\n"
-"        dst[tid] += src[j];\n"
-"    }\n"
-"\n"
-"}\n";
-
-
-int
-verify_loop(int *inptr, int *loopindx, int *loopcnt, int *outptr, int n)
+namespace {
+const char *loop_kernel_code = R"(
+__kernel void test_loop(__global int *src, __global int *loopindx, __global int *loopcnt, __global int *dst)
 {
-    int     r, i, j, k;
+    int  tid = get_global_id(0);
+    int  n = get_global_size(0);
+    int  i, j;
 
-    for (i=0; i<n; i++)
+    dst[tid] = 0;
+    for (i=0, j=loopindx[tid]; i<loopcnt[tid]; i++, j++)
+    {
+        if (j >= n)
+            j = 0;
+        dst[tid] += src[j];
+    }
+}
+)";
+
+
+int verify_loop(std::vector<cl_int> inptr, std::vector<cl_int> loopindx,
+                std::vector<cl_int> loopcnt, std::vector<cl_int> outptr, int n)
+{
+    int r, i, j, k;
```

Comment:
Can we move these variable declaration to where they are initialized? e.g `for (int i = 0;...`. This is quite a nice clean-up when you're done elsewhere.